### PR TITLE
Clarify `kubectl get service` output.

### LINKED ISSUE #24411: Clarify `kubectl get service` output
Someone reported:
```
I then use kubectl get services and I don’t have an external IP address. 
NAME         CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
frontend     10.163.255.126                 80/TCP    24s
kubernetes   10.163.240.1     <none>        443/TCP   27m

I panic a little cause I thought I requested it. I look up a few things and in a lark rerun and kubectl get services get:
NAME         CLUSTER-IP       EXTERNAL-IP     PORT(S)   AGE
frontend     10.163.255.126   146.X.Y.Z   80/TCP    4m
kubernetes   10.163.240.1     <none>          443/TCP   32m

Okay. It might be nice if it had told me that it was requesting, instead of just being blank.
```

The empty output is caused by:
```
func getServiceExternalIP(svc *api.Service) string {
	switch svc.Spec.Type {
	case api.ServiceTypeClusterIP:
		if len(svc.Spec.ExternalIPs) > 0 {
			return strings.Join(svc.Spec.ExternalIPs, ",")
		}
		return "<none>"
	case api.ServiceTypeNodePort:
		if len(svc.Spec.ExternalIPs) > 0 {
			return strings.Join(svc.Spec.ExternalIPs, ",")
		}
		return "nodes"
	case api.ServiceTypeLoadBalancer:
		lbIps := loadBalancerStatusStringer(svc.Status.LoadBalancer)
		if len(svc.Spec.ExternalIPs) > 0 {
			result := append(strings.Split(lbIps, ","), svc.Spec.ExternalIPs...)
			return strings.Join(result, ",")
		}
		return lbIps
	}
	return "unknown"
}
```

Note the last case doesn't check for an empty list. 

We should settle on a term for values that are currently empty but we expect to be populated asynchronously. I don't like "requesting". Elsewhere we use "pending", such as for unscheduled pods, so let's go with that. 

And, let's ensure that all the non-literal values are bracketed consistently: `<pending>`, `<nodes>`, `<unknown>`.

We should update the kubectl conventions accordingly.

Similar recent PRs: #22279,  #22635

cc @AdoHe

### PR DESCRIPTION
**Pull request by k82cn**
fixes #24411 .

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/24475)

### CODE REVIEW

**Comment by jlowdermilk** [comment]
cc @janetkuo for output style

**Comment by jlowdermilk** [comment]
@k8s-bot ok to test

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -876,16 +876,19 @@ func getServiceExternalIP(svc *api.Service) string {
...
 		lbIps := loadBalancerStatusStringer(svc.Status.LoadBalancer)
 		if len(svc.Spec.ExternalIPs) > 0 {
 			result := append(strings.Split(lbIps, ","), svc.Spec.ExternalIPs...)
 			return strings.Join(result, ",")
 		}
-		return lbIps
+		if len(lbIps) > 0 {
+			return lbIps
+		}
+		return "<none>"
```
Comment (janetkuo) [inline pkg/kubectl/resource_printer.go]: What if it's just pending?

**Comment by bgrant0607** [comment]
cc @AdoHe

File: pkg/kubectl/resource_printer.go (hunk 1 above)
Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: Clarification: This should be `<pending>`, not `<none>`.

Comment (k82) [inline pkg/kubectl/resource_printer.go]: Thanks for your comments :). Let me double check the related code for this patch. I'll update the patch accordingly.

**Comment by AdoHe** [comment]
@k82 please update the test case and make sure it's ok.
```
FAIL	k8s.io/kubernetes/pkg/kubectl/cmd	5.611s
```

**Comment by k82** [comment]
@AdoHe , sure. I found this failed test :). I'll also correct it.

**Comment by k82** [comment]
Re-open it for latest patch.

**Comment by k82** [comment]
is this patch OK to mark "LGTM"? If any more comments, please let me know :).

**Pull request merged**